src/main/java/Storage.java

```diff
@@ -0,0 +1,136 @@
+                if (addLine[1].equals("1")) {
+                    t.setDone();
+                }
+                list.add(t);
+            } else if (addLine[0].equals("D")) {
+                Deadline d = new Deadline(addLine[2], addLine[3]);
+                if (addLine[1].equals("1")) {
+                    d.setDone();
+                }
+                list.add(d);
+            } else if (addLine[0].equals("E")) {
+                Events e = new Events(addLine[2], addLine[3]);
+                if (addLine[1].equals("1")) {
+                    e.setDone();
+                }
+                list.add(e);
+            }
+        }
+        return list;
+    }
+
+    public void updateTask(int doneTask, TaskList tasks) throws IOException {
+        List<Task> list = tasks.getList();
+        FileReader fr = new FileReader(filepath);
+        BufferedReader br = new BufferedReader(fr);
+        List<String> tempArr = new ArrayList<String>();
+        for (int i = 0; i < list.size(); i++) {
+            if (i == doneTask) {
+                String oldLine = br.readLine();
+                String updatedLine = oldLine.replaceFirst("0", "1");
+                tempArr.add(updatedLine);
+            } else {
+                tempArr.add(br.readLine());
+            }
+        }
+        Files.delete(Paths.get(filepath));
+        File newFile = new File(filepath);
+        FileWriter fw = new FileWriter(filepath);
+        for (int j = 0; j < tempArr.size(); j++) {
+            if (j == 0) {
+                fw.write(tempArr.get(j));
+            } else {
+                fw.write("\n");
+                fw.write(tempArr.get(j));
+            }
+        }
+        fw.close();
+    }
+
+    public void deleteTask(int deleteTask, TaskList tasks) throws IOException {
+        List<Task> list = tasks.getList();
+        FileReader fr = new FileReader(filepath);
+        BufferedReader br = new BufferedReader(fr);
+        List<String> tempArr = new ArrayList<String>();
+        for (int i = 0; i < list.size(); i++) {
+            if (i == deleteTask) {
+                String delete = br.readLine();
+                continue;
+            } else {
+                tempArr.add(br.readLine());
+            }
+        }
+        Files.delete(Paths.get(filepath));
+        File newFile = new File(filepath);
+        FileWriter fw = new FileWriter(filepath);
+        for (int j = 0; j < tempArr.size(); j++) {
+            if (j == 0) {
+                fw.write(tempArr.get(j));
+            } else {
+                fw.write("\n");
+                fw.write(tempArr.get(j));
+            }
+        }
+        fw.close();
+    }
+
+    public void addTodo(TaskList tasks) throws IOException {
+        List<Task> list = tasks.getList();
+        Task t = list.get(list.size() - 1);
+        FileWriter fw = new FileWriter(filepath, true);
+        fw.write("\n");
+        fw.write("T@" + t.getStatusNumber() + "@" + t.getDescription());
```

Comment:
This can be abstracted out for example, together with the respective line in addDeadline and addEvent